File: python/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        char_count = defaultdict(lambda: 0)
        for c in s:
            char_count[c] += 1
        for i in range(len(s)):
            if char_count[s[i]] == 1:
                return i
        return -1

File: rust/first-unique-character-in-a-string.rs
use std::collections::HashMap;

impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut char_count = HashMap::new();
        for c in s.chars() {
            *char_count.entry(c).or_insert(0) += 1;
        }
        let mut i = 0;
        for c in s.chars() {
            if char_count[&c] == 1 {
                return i;
            }
            i += 1;
        }
        return -1
    }
}
